src/processors/vestingContractProcessor.ts

```diff
@@ -114,79 +104,77 @@ export const processVestingContract = async (
-      const errMessage = res.value.err.toString();
-      if (!errMessage.includes("BlockhashNotFound") || i === 2) {
-        throw new Error(errMessage);
-      }
+  while (true) {
+    const contractId = await fetchExistingStream(connection, pid, mint, recipientInfo.address);
+    if (contractId) {
+      console.log(`Recipient ${recipientInfo.address.toBase58()} already has a stream for this mint, will skip`);
+      return { txId: "", contractId: contractId };
     }
-    break;
-  }
 
-  let signature = bs58.encode(tx.signatures[0]);
-  let blockheight = await connection.getBlockHeight(commitment);
-  const rawTransaction = tx.serialize();
-  let transactionSent = false;
-  while (blockheight < recentBlockInfo.lastValidBlockHeight + 15) {
-    try {
-      if (blockheight < recentBlockInfo.lastValidBlockHeight || !transactionSent) {
-        await SEND_THROTTLE(async () => {
-          await connection.sendRawTransaction(rawTransaction, {
-            maxRetries: 0,
-            minContextSlot: context.slot,
-            preflightCommitment: commitment,
-            skipPreflight: true,
-          })
-        });
-        transactionSent = true;
-      }
-    } catch (e) {
-      if (
-        transactionSent ||
-        (e instanceof SendTransactionError && e.message.includes("Minimum context slot has not been reached"))
-      ) {
-        await sleep(500);
-        continue;
-      }
-      throw e;
+    const { context, value: recentBlockInfo } = await connection.getLatestBlockhashAndContext({ commitment });
+    let tx = buildStreamTransaction(ixs, recentBlockInfo.blockhash, sender, metadata);
+    const res = await simulateTransaction(connection, tx);
+    let newCu = DEFAULT_CU;
+    if (res.value.unitsConsumed) {
+      newCu = Math.floor(res.value.unitsConsumed * (1 + COMPUTE_ERROR_MARGIN / 10_000));
     }
-    await sleep(500);
-    try {
-      const value = await confirmAndEnsureTransaction(connection, signature);
-      if (value) {
-        return { txId: signature, contractId: metadata.publicKey.toBase58() };
+    ixs[0] = ComputeBudgetProgram.setComputeUnitLimit({ units: newCu });
+    tx = buildStreamTransaction(ixs, recentBlockInfo.blockhash, sender, metadata);
 
-      }
-    } catch (e) {
-      if (e instanceof TransactionFailedError) {
+    let signature = bs58.encode(tx.signatures[0]);
+    let blockheight = await connection.getBlockHeight(commitment);
+    const rawTransaction = tx.serialize();
+    let transactionSent = false;
+    while (blockheight < recentBlockInfo.lastValidBlockHeight + 15) {
+      try {
+        if (blockheight < recentBlockInfo.lastValidBlockHeight || !transactionSent) {
+          await SEND_QUEUE.add(async () => {
+            await connection.sendRawTransaction(rawTransaction, {
+              maxRetries: 0,
+              minContextSlot: context.slot,
+              preflightCommitment: commitment,
+              skipPreflight: true,
+            });
+          });
```

Comment:
Accordingly to the docs, `.add` accepts a function that returns a Promise
Imo it is better to be more explicit here and actually return values for it

```suggestion
          await SEND_QUEUE.add(() => connection.sendRawTransaction(rawTransaction, {
              maxRetries: 0,
              minContextSlot: context.slot,
              preflightCommitment: commitment,
              skipPreflight: true,
            })
          );
```


Additionally from docs:
```
If your items can potentially throw an exception, you must handle those errors from the returned Promise or they may be reported as an unhandled Promise rejection and potentially cause your process to exit immediately.
```
and `sendRawTransaction` can. Do we need to handle it?